test/API/H5_api_test_util.c

```diff
@@ -797,3 +935,223 @@ remove_test_file(const char *prefix, const char *filename)
+            fprintf(stderr, "Unable to copy VOL connector string\n");
+            goto error;
+        }
+
+        if (NULL == (vol_connector_name = (const char*) strtok(vol_connector_string_copy, " "))) {
+            fprintf(stderr, "Error while parsing VOL connector string\n");
+            goto error;
+        }
+
+    }
+
+    /* If VOL is not native, make sure it is registered properly */
+    if (0 != strcmp(vol_connector_name, "native")) {
+        htri_t is_registered;
+
+        if ((is_registered = H5VLis_connector_registered_by_name(vol_connector_name)) < 0) {
+            fprintf(stderr, "Unable to determine if VOL connector is registered\n");
+            goto error;
+        }
+
+        if (!is_registered) {
+            fprintf(stderr, "Specified VOL connector '%s' wasn't correctly registered!\n",
+                    vol_connector_name);
+            goto error;
+        }
+        else {
+            /*
+             * If the connector was successfully registered, check that
+             * the connector ID set on the default FAPL matches the ID
+             * for the registered connector before running the tests.
+             */
+            if ((fapl_id = H5Pcreate(H5P_FILE_ACCESS)) < 0) {
+                fprintf(stderr, "Couldn't create FAPL\n");
+                goto error;
+            }
+
+            if (H5Pget_vol_id(fapl_id, &default_con_id) < 0) {
+                fprintf(stderr, "Couldn't retrieve ID of VOL connector set on default FAPL\n");
+                goto error;
+            }
+
+            if ((registered_con_id = H5VLget_connector_id_by_name(vol_connector_name)) < 0) {
+                fprintf(stderr, "Couldn't retrieve ID of registered VOL connector\n");
+                goto error;
+            }
+
+            if (default_con_id != registered_con_id) {
+                fprintf(stderr, "VOL connector set on default FAPL didn't match specified VOL connector\n");
+                goto error;
+            }
+        }
+    }
+
+    if (registered_con_id > 0 && H5VLclose(registered_con_id) < 0) {
+        fprintf(stderr, "Unable to close registered VOL connector\n");
+        goto error;
+    }
+
+    if (default_con_id > 0 && H5VLclose(default_con_id) < 0) {
+        fprintf(stderr, "Unable to close default VOL connector\n");
+        goto error;
+    }
+
+    if (fapl_id > 0 && H5Pclose(fapl_id) < 0) {
+        fprintf(stderr, "Unable to close FAPL\n");
+        goto error;
+    }
+
+    free(vol_connector_string_copy);
+
+    return 0;
+
+error:
+    H5E_BEGIN_TRY {
+        H5VLclose(registered_con_id);
+        H5VLclose(default_con_id);
+        H5Pclose(fapl_id);
+        free(vol_connector_string_copy);
+    } H5E_END_TRY;
+
+    return -1;
+}
+
+void H5_api_test_display_results(void) {
+    const char *vol_connector_name = NULL;
+
+    if (NULL == (vol_connector_name = HDgetenv(HDF5_VOL_CONNECTOR)))
+        vol_connector_name = "native";
```

Comment:
Separating this logic from the logic now in `H5_api_test_display_information` and `H5_api_check_vol_registration` means the VOL connector name will accidentally include the info string specified, if any, and the string has to be parsed multiple times to get the same result. I think all of the helper routine refactoring changes should be put into a separate PR from this one and thought about a bit more. The helper routines are nice for bringing down the size of main, but it's really unrelated to what this PR is trying to do.